examples/get_matching/server.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (C) Bull S.A.S - 2024
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ * BXI Low Level Team
+ *
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "portals4.h"
+#include "portals4_bxiext.h"
+
+#include "get_me.h"
+
+/*
+ * this file represent a server preparing messages which can be get from a client
+ *
+ * Usage: ./server [max messages possible to get]
+ *
+ * The server will exit after the specified number of messages have been got.
+ * By default, the server wait forever.
+ */
+
+int main(int argc, char **argv)
+{
+	int res;
+	int ret;
+	ptl_handle_ni_t nih;
+	ptl_index_t pti;
+	ptl_handle_eq_t eqh;
+	ptl_process_t id;
+	ptl_event_t ev;
+	ptl_handle_me_t meh;
+	int n = 1;
+	char msg[PTL_EV_STR_SIZE];
+	ptl_process_t id_server = { .phys = { .pid = SERVER_PID, .nid = 1 } };
+	char data[11];

```

Comment:
Let's put a bit more than 1 message in the ME. For example we can do `char data[MESSAGELENGTH * 10];`, and later:

```C
for (int i = 0; i < 10; i++)
    sprintf(&data[i * MESSAGELENGTH], "message%d !", i);
```